lib/train-habitat/connection.rb

```diff
@@ -0,0 +1,44 @@
+require 'net/http'
+require 'json'
+require 'train-habitat/platform'
+
+module TrainPlugins
+  module Habitat
+    class Connection < Train::Plugins::Transport::BaseConnection
+      include TrainPlugins::Habitat::Platform
+
+      def initialize(options)
+        msg = 'Habitat host serving HTTP Gateway is required'
+        raise Train::TransportError, msg if options.nil? || options[:host].nil?
```

Comment:
`raise Train::TransportError, msg if options&.[:host].nil?`

Reply:
Actually, `msg` is only used for this, so I would make it a conditional structure instead. No local variable.
```ruby
if options&.[:host].nil?
  raise Train::TransportError, 'Habitat host serving HTTP Gateway is required'
end
```